Makefile

```diff
@@ -280,9 +316,42 @@ check-bootstrap:
 		-k ${BOOTSTRAP_KEY} \
 		-r ${BOOTSTRAP_REPO}
 
-authget = tok=$$(chainctl auth token --audience=$(1)) || \
+define authget
+  tok=$$(chainctl auth token --audience=$(1)) || \
   { echo "failed token from $(1) for target $@"; exit 1; }; \
   mkdir -p $$(dirname $(2)) && \
   echo "auth-download[$(1)] to $(2) from $(3)" && \
   curl --fail -LS --silent -o $(2).tmp --user "user:$$tok" $(3) && \
-	mv "$(2).tmp" "$(2)"
+  mv "$(2).tmp" "$(2)"
+endef
+
+define repo_token_if_needed
+  $(eval yamlfile := $1.yaml)
+  $(eval pkgname := $1)
+  # iamguarded pipelines use auth/guarded-repo
+  $(eval pipelines := "auth/guarded-repo|iamguarded")
+  if yq e -e '.. | select(has("uses")) \
+                 | select(.uses | test($(pipelines)))' \
+                 $(yamlfile) > /dev/null 2>&1; then \
+    echo "Creating guarded repo token for $(pkgname)…"; \
+    $(MAKE) repo-token/$(pkgname); \
+  else \
+    echo "$1 does not use a guarded repo. Skipping token creation."; \
+  fi
+endef
+
+define libraries_token_if_needed
+  $(eval yamlfile := $1.yaml)
+  $(eval pkgname := $1)
+  $(eval pipelines := "auth")
+  $(eval ignore_pipelines := "auth/guarded-repo")
+  if yq e -e '.. | select(has("uses")) \
```

Comment:
btw, is `yq` used before this in our Makefiles? If not, do you think the error mode for not having it installed is clear, or do we need an explicit message? (I haven't tried it)